ipyleaflet/leaflet.py

```diff
@@ -579,7 +579,7 @@ def _get_data(self):
             elif self.data['type'] == 'FeatureCollection':
                 for feature in self.data['features']:
                     feature['properties']['style'] = self.style_callback(feature)
-        else:
+        elif self.style:
             if self.data['type'] == 'Feature':
                 self.data['properties']['style'] = self.style
```

Comment:
In the method, we should replace all the `self.data['properties']['style'] = something` into `self.data['properties']['style'].update(something)`

Reply:
this way we never overwrite the style, we simply update it.

Reply:
But if the style key is not present within the `feature` then will it give a `KeyError`

Reply:
Made changes as suggested :)